sessions/2018/november/Session_17.md

```diff
@@ -0,0 +1,79 @@
+
+##### Domain Specific Language   
+
+Domain Specific Languages (DSL) are small programming languages used inside an application with the purpose to execute certain focused tasks into applications, examples of them are SQL, CSS, Regular Expressions. Gradle scripts written with Groovy are other exampke if a DLS, that means an DLS can be written in another programming language that isn't a DSL.
+
+##### Java vs JavaScript comparison
+
+Was stated that Java produce much more enterprise ready applications because its multithreading support.
+
+Very good things can been achieved with JavaScript in the backend and the fronted -besides of the fact to be using the same programming language for both of them- but with JavaScript there is a need that the programmer deal beforehand with the problems can arise with a single threaded JavaScript application, such as blocking and long time of response when making operations. In general callback functions (that could leads to what is known as [callback hell][9]), [promises][7], [run-parallel][12] and [run-series][13] are used as alternatives to overcome the breaking application risks (whitout be a solution for single threading), but additional to that, we should too think in horizontal scaling and load balancing to have a enterprise grade application that behaves similarly as a backend Java application.
+
+So, at the end Java is simpler and safer to use to build stable backend applications.
+
+
+### Homework
+Make the fronted for the Java API calculator.
+
+### Resources
+
+- [Apache Groovy Language][11]
+- [Domain-specific language][1]
+- [Domain Specific Languages Book][2]
+- [Domain Specific Languages Video][3]
+- [JavaScript vs CofeeScript vs TypeScript][4]
+- [CofeeScript][10]
+- [Coffeescript vs. Javascript: Dog eat Dog][5]
+- [A Case Against Using CoffeeScript][6]
+- [Understanding promises in JavaScript][7]
+- [Implementing Promises In JavaScript][8]
+- [Callback Hell][9]
+- [run-parallel][12]
+- [run-series][13]
+
+
+[1]: https://martinfowler.com/books/dsl.html
+[2]: https://martinfowler.com/dsl.html
+[3]: https://channel9.msdn.com/blogs/charles/expert-to-expert-martin-fowler-and-chris-sells-perspectives-on-domain-specific-languages
+[4]: https://bytescout.com/blog/2016/07/javascript-vs-coffeescript-vs-typescript.html
+[5]: https://lostechies.com/bradcarleton/2013/10/23/coffeescript-vs-javascript-dog-eat-dog/
+[6]: http://ryanflorence.com/2011/case-against-coffeescript/
```

Comment:
Ryan Florence to is a very skilled engineer - if you're interested in React (a very popular JavaScript library+ecosystem) and use Twitter, I recommend following him.